lib/events/dynamoevents/dynamoevents.go

```diff
@@ -1566,12 +1584,12 @@ func (l *eventsFetcher) processQueryOutput(output *dynamodb.QueryOutput, hasLeft
 	l.checkpoint.Iterator = ""
 
 	if output.LastEvaluatedKey != nil {
-		m := make(map[string]any)
-		if err := attributevalue.UnmarshalMap(output.LastEvaluatedKey, &m); err != nil {
+		var e event
+		if err := attributevalue.UnmarshalMap(output.LastEvaluatedKey, &e); err != nil {
```

Comment:
Can we use a separate type that only includes the fields of the key rather than the whole event?